generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/GeneratedClassWriter.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.justice.services.generators.commons.helper;
+
+
+import static java.lang.String.format;
+import static uk.gov.justice.services.generators.commons.helper.Names.JAVA_FILENAME_SUFFIX;
+import static uk.gov.justice.services.generators.commons.helper.Names.RESOURCE_PACKAGE_NAME;
+
+import uk.gov.justice.raml.core.GeneratorConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.TypeSpec;
+import org.slf4j.Logger;
+
+public class GeneratedClassWriter {
+
+    private GeneratedClassWriter() {}
+
+    public static void writeClass(final GeneratorConfig configuration, final String packageName, final TypeSpec typeSpec, final Logger logger) {
+        try {
+            if (classDoesNotExist(configuration, typeSpec)) {
+                JavaFile.builder(packageName, typeSpec)
+                        .build()
+                        .writeTo(configuration.getOutputDirectory());
+            } else {
+                logger.warn("The class {} already exists, skipping code generation.", typeSpec.name);
+            }
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static boolean classDoesNotExist(final GeneratorConfig configuration, final TypeSpec typeSpec) {
```

Comment:
Would prefer this naming to not be in the negative, so classExists as the method name, but maybe just a personal thing...